### PR TITLE
feature: add podLabels to nats helmchart

### PR DESCRIPTION
**Pull request by icelynjennings**
Allow templating podLabels on NATS statefulset and nats box.

Relevant issues: #298

### CODE REVIEW

File: helm/charts/nats/values.yaml [hunk 1]
```diff
@@ -229,6 +229,12 @@ tolerations: []
 # Annotations to add to the NATS StatefulSet
 statefulSetAnnotations: {}
 
+# Labels to add to the NATS StatefulSet
+statefulSetLabels: {}
```
Comment (1995parham) [inline helm/charts/nats/values.yaml:233]: we have common labels for this but by this, you can have specific labels for stateful sets do you need this?

File: helm/charts/nats/values.yaml [hunk 2]
```diff
@@ -229,6 +229,12 @@ tolerations: []
...
+
+# Labels to add to the pods of the NATS StatefulSet
+statefulSetPodLabels: {}
```
Comment (1995parham) [inline helm/charts/nats/values.yaml:236]: we can use podLabel for this.

File: helm/charts/nats/values.yaml (hunk 1 above)
Comment (icelynjennings) [reply helm/charts/nats/values.yaml]: You're right - reverted.

File: helm/charts/nats/values.yaml (hunk 2 above)
Comment (icelynjennings) [reply helm/charts/nats/values.yaml:233]: This is to differentiate between nats statefulset pods and natsbox deployment pods

**Review by wallyqs — APPROVED** [review]
LGTM, let's give this a try.

**Pull request merged**